### ISSUE TITLE
Plugin settings shouldn't get written to $CFG

### ISSUE THREAD

**Issue by Peterburnett** [blindsidenetworks/moodle-mod_bigbluebuttonbn#267]
Moodle has an API to force plugin config in $CFG using $CFG->forced_plugin_settings, which can be used in place of directly writing to $CFG. It may require additional work to hide the forced config from the Admin GUI as the current implementation does, but will allow for the entire plugin config to be grabbed in a single get_config() and accessed, making the code much more readable.

**Comment by jfederico** [comment]
Yeah, agree. It has to be reviewed/refactored. The thing is that this impllementation was done in this plugin before Moodle had something to cover what we did. We are talking of 2014 easily and I remember talking about this with Martin in a MoodleMoot in Montreal :)

Eitherway, I would be open for a contribution to fix it if you are willing to volunteer :)

**Comment by brendanheywood** [comment]
See also https://tracker.moodle.org/browse/CONTRIB-8399
